### PR TITLE
Adds nice provisioning progress to Artemis provision plugin

### PR DESCRIPTION
**Pull request by happz**
Artemis plugin now displays the sequence of of guest states
as its provisioning progresses. This has been implemented as
a reusable primitive context manager, coupled with a small
refactoring of code taking care of indentation.

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -1780,3 +1811,78 @@ def get_distgit_handler(remotes=None, usage_name=None):
 def get_distgit_handler_names():
     """ All known distgit handlers """
     return [i.usage_name for i in DistGitHandler.__subclasses__()]
+
+
+class updatable_message(contextlib.AbstractContextManager):
```
Comment (lukaszachy) [inline tmt/utils.py:1816]: Shouldn't class name start with upper case?

**Review by lukaszachy — COMMENTED** [review]
One nitpick about class name, otherwise looks very good. Docstrings and typing made understanding code much easier.

File: tmt/utils.py (hunk 1 above)
Comment (happz) [reply tmt/utils.py:1816]: I'd say being used as a context manager pushes the first letter into the lowercase territory. Similar to `open`, `socket.create_connection`, `urllib.request.urlopen` (although those are not classes, just context managers), (very soft) naming convention seems to be in favor of context managers and decorators following function naming convention.

And `with UpdatableMessage('state', indent_level=self._level()) as progress_message:` looks just weird :)

Comment (psss) [reply tmt/utils.py:1817]: Yeah, for context manager it makes sense to have it lower case.

**Review by psss — APPROVED** [review]
Thanks for the improvement. Looks (and works) very nice! Just adjusted a couple of line breaks in 4b52d0a. Going to squash and merge.

**Pull request merged**